src/adam7.rs

```diff
@@ -88,16 +162,16 @@ impl Adam7Iterator {
     fn init_pass(&mut self) {
         let w = f64::from(self.width);
         let h = f64::from(self.height);
-        let (line_width, lines) = match self.current_pass {
-            1 => (w / 8.0, h / 8.0),
-            2 => ((w - 4.0) / 8.0, h / 8.0),
-            3 => (w / 4.0, (h - 4.0) / 8.0),
-            4 => ((w - 2.0) / 4.0, h / 4.0),
-            5 => (w / 2.0, (h - 2.0) / 4.0),
-            6 => ((w - 1.0) / 2.0, h / 2.0),
-            7 => (w, (h - 1.0) / 2.0),
-            _ => unreachable!(),
-        };
+
+        let info = PassConstants::PASSES[self.current_pass as usize - 1];
+        // Note: integer arithmetic `div_ceil` would not handle the cases `w < 8` where the
+        // intermediate value becomes negative. Since offset <= 0.5 * sampling we get the correct
+        // ceiling value with floating point arithmetic however. Also note the float arithmetic is
+        // exact since all divisions are pure exponent offsets and the mantissa holds any possible
+        // `[-8; 0) + u32[]` value.
```

Comment:
I think it should work if we use `saturating_sub` and `div_ceil` together: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=17f92086d15d1db125435913458c5c02

Reply:
Since this isn't a performance sensitive path I'd rather prefer the readability. Just ran it on the whole range of `0..=u32::MAX` just to be sure, and it checks out. I'll change that.

Reply:
Note: I found why we're not using that yet, but we can easily bump the MSRV to `1.73`, as required, in the `0.18` release series.